src/Components/LoginComponent/Signup.js

```diff
@@ -0,0 +1,319 @@
+import React, { useState } from "react";
+import styled, { ThemeProvider } from "styled-components";
+import { withRouter } from "react-router-dom";
+import Apiresource from "../../Config";
+
+
+function Signup(props, visible) {  
+    const [ userInfo, setName ] = useState({email:"",password:"",name:""});
+
+
+    const emailDetector = (e) => {
+        setName({...userInfo, [e.target.name]:e.target.value});
+        }; 
+
+
+    const submit = (userInfo) => {
+        fetch(`${Apiresource}/account/sign-up`, {
+            method: 'POST',
+            body: JSON.stringify({
+                email: userInfo.email,
+                password: userInfo.password,
+                name : userInfo.nickname
+            }),
+        })
+        .then(res => res.json)
+        .then(res => {
+            if (res) {
+                alert("회원가입을 환영합니다")
+                props.history.push('/main')
+            } else {
+                alert('이메일과 비밀번호를 확인해주십시오')
+            };
+        })
+    }
+
+    const returnSocialLogin = () => {
+        props.history.push('/SocialLogin')
+    }
+
+    const returnLogin = () => {
+        props.history.push('/EmailLogin')
+    }
+
+    const returnMain = () => {
+        props.history.push('/Main')
+    }
+
+    const word = '@';
+    const isValidEmail = userInfo.email.length && !(userInfo.email.includes(word));
```

Comment:
```suggestion
    const isValidEmail = userInfo.email.length && userInfo.email.includes(word);
```
느낌표를 넣어주면 @가 찾아졌을 때 -> false가 되지 않나요?

Reply:
&&때문에 !를 붙인 경우가 아닌경우에 작동되는 것으로 알고 있는데, 제가 잘못알고 있는건가요?????

Reply:
위 로직을 하게 되면 이메일에 @이 있을 때 false가 되버립니다!